frankenphp.go

```diff
@@ -191,7 +191,7 @@ func NewRequestWithContext(r *http.Request, opts ...RequestOption) (*http.Reques
 	// SCRIPT_FILENAME is the absolute path of SCRIPT_NAME
 	fc.scriptFilename = sanitizedPathJoin(fc.documentRoot, fc.scriptName)
 
-	return r.WithContext(context.WithValue(r.Context(), contextKey, fc)), nil
+	return r.WithContext(context.WithValue(context.Background(), contextKey, fc)), nil
```

Comment:
Couldn't you tweak the context to pass in the Caddy module instead? It may have legit use cases to pass a custom context here.

Reply:
Moved the PR to WIP as I'm on the tails of a much better solution. It appears we are holding a reference to the Caddy request somehow and that is the crux of the memory leak. Essentially, every request object stays in memory forever and ever in worker mode.

I'm about halfway to narrowing down the leak, but I won't be able to get back to this until at least Wednesday/Thursday.